packages/x-charts-pro/src/FunnelChart/formatter.ts

```diff
@@ -0,0 +1,157 @@
+}) =>
+  inverse
+    ? { x: other, y: main, useBandWidth, stackOffset }
+    : { x: main, y: other, useBandWidth, stackOffset };
+
+const formatter: SeriesProcessor<'funnel'> = (params, dataset) => {
+  const { seriesOrder, series } = params;
+
+  // Create a data set with format adapted to d3
+  const d3Dataset: FunnelDataset = (dataset as FunnelDataset) ?? [];
+  seriesOrder.forEach((id) => {
+    const data = series[id].data;
+    if (data !== undefined) {
+      data.forEach((item, index) => {
+        if (d3Dataset.length <= index) {
+          d3Dataset.push({ [id]: item.value });
+        } else {
+          d3Dataset[index][id] = item.value;
+        }
+      });
+    } else if (dataset === undefined) {
+      throw new Error(
+        [
+          `MUI X: funnel series with id='${id}' has no data.`,
+          'Either provide a data property to the series or use the dataset prop.',
+        ].join('\n'),
+      );
+    }
+  });
+
+  const completedSeries: Record<string, ChartSeriesDefaultized<'funnel'>> = {};
+
+  seriesOrder.forEach((seriesId) => {
+    const currentSeries = series[seriesId];
+    // Use dataKey if needed and available
+    // const dataKey = currentSeries.dataKey;
+    const dataKey = undefined;
+
+    const isHorizontal = currentSeries.layout === 'horizontal';
+
+    completedSeries[seriesId] = {
+      labelMarkType: 'square',
+      layout: isHorizontal ? 'horizontal' : 'vertical',
+      valueFormatter: (item) => (item == null ? '' : item.value.toLocaleString()),
+      ...currentSeries,
+      data: dataKey
+        ? dataset!.map((data, i) => {
+            const value = data[dataKey];
+            if (typeof value !== 'number') {
+              if (process.env.NODE_ENV !== 'production') {
+                warnOnce([
+                  `MUI X: your dataset key "${dataKey}" is used for plotting funnels, but contains nonnumerical elements.`,
+                  'Funnel plots only support number values.',
+                ]);
+              }
+              return { id: `${seriesId}-funnel-item-${i}`, value: 0 };
+            }
+            return { id: `${seriesId}-funnel-item-${i}`, value };
+          })
+        : currentSeries.data!.map((v, i) => ({ id: `${seriesId}-funnel-item-${v.id ?? i}`, ...v })),
+      dataPoints: [],
+    };
+
+    const stackOffsets = completedSeries[seriesId].data
+      .toReversed()
+      .map((_, i, array) => array.slice(0, i).reduce((acc, item) => acc + item.value, 0))
+      .toReversed();
+
+    completedSeries[seriesId].dataPoints = completedSeries[seriesId].data.map(
+      (item, dataIndex, array) => {
+        const currentMaxMain = item.value ?? 0;
+        const nextDataIndex = dataIndex === array.length - 1 ? dataIndex : dataIndex + 1;
+        const nextMaxMain = array[nextDataIndex].value ?? 0;
+        const nextMaxOther = 0;
+        const currentMaxOther = completedSeries[seriesId].data[dataIndex].value;
+        const stackOffset = stackOffsets[dataIndex];
```

Comment:
I assume the idea is to latter add options here to allow for example customizing the logic for current/next choice